spec/delayed/job_spec.rb

```diff
@@ -253,29 +253,112 @@ def create_job(opts = {})
   end
 
   describe '#name' do
-    it 'is the class name of the job that was enqueued' do
-      expect(described_class.create(payload_object: ErrorJob.new).name).to eq('ErrorJob')
-    end
+    context 'when name column is populated' do
```

Comment:
I split into two three contexts (when the name column exists and is populated, when it exists and is _not_ populated, and when it doesn't exist at all).

In a subsequent release, we can simplify this with the assumption that it exists and is not nullable.